Packages/com.llealloo.audiolink/Runtime/Scripts/YtdlpPlayer.cs

```diff
@@ -0,0 +1,303 @@
+                if(doDownload)
+                    DownloadYtdlp();
+
+                if(!Application.isPlaying)
+                    EditorApplication.ExitPlaymode();
+                
+                #elif UNITY_EDITOR_LINUX
+                    EditorUtility.DisplayDialog("[AudioLink] Missing yt-dlp", "try \"sudo pacman -S yt-dlp\"", "ok");
+                #endif
+            }
+
+            if (!File.Exists(_ytdlpPath)) 
+            {
+                // Still don't have it, no dice
+                Debug.LogWarning("[AudioLink] Unable to find yt-dlp");
+                return;
+            }
+            else
+            {
+                // Found it
+                _ytdlFound = true;
+                Debug.Log($"[AudioLink] Found yt-dlp at path '{_ytdlpPath}'");
+            }
+        }
+
+        /// <summary> Resolves a URL to one usable in a VideoPlayer. </summary>
+        /// <param name="url">URL to resolve for playback</param>
+        /// <param name="resolution">Resolution (vertical) to request from yt-dlp</param>
+        public static string Resolve(string url, int resolution)
+        {
+            if(!_ytdlFound)
+            {
+                Debug.LogWarning($"[AudioLink] Unable to resolve URL '{url}' : yt-dlp not found");
+                return null;
+            }
+
+            Process ytdlProcess = new Process();
+
+            ytdlProcess.StartInfo.WindowStyle = ProcessWindowStyle.Hidden;
+            ytdlProcess.StartInfo.CreateNoWindow = true;
+            ytdlProcess.StartInfo.UseShellExecute = false;
+            ytdlProcess.StartInfo.RedirectStandardOutput = true;
+            ytdlProcess.StartInfo.FileName = _ytdlpPath;
+            ytdlProcess.StartInfo.Arguments = $"--no-check-certificate --no-cache-dir --rm-cache-dir -f \"mp4[height<=?{resolution}]/best[height<=?{resolution}]\" --get-url \"{url}\"";
+
+            Debug.Log($"[AudioLink] Attempting to resolve URL '{url}'");
+
+            try
+            {
+                ytdlProcess.Start();
+                _runningYtdlProcesses.Add(ytdlProcess);
+
+                while (!ytdlProcess.HasExited)
+                    new WaitForSeconds(0.1f);
+
+                _runningYtdlProcesses.Remove(ytdlProcess);
+
+                return ytdlProcess.StandardOutput.ReadLine();
+            }
+            catch(Exception e)
+            {
+                Debug.LogWarning($"[AudioLink] Unable to resolve URL '{url}' : " + e.Message);
+                return null;
+            }
+        }
+
+        /// <summary> Resolves a URL and set a videoplayer's URL to the resolved URL. </summary>
+        /// <param name="url">URL to resolve for playback</param>
+        /// <param name="resolution">Resolution (vertical) to request from yt-dlp</param>
+        /// <param name="player">VideoPlayer component to set URL on</param>
+        public static void ResolveAndSet(string url, int resolution, VideoPlayer player)
+        {
+            try
```

Comment:
this try/catch is pointless, you already handle exceptions in Resolve

Reply:
Removed, resolved in 017fd35fa743a75b5ee7b0914fd4a2328bfc13d4